src/pages/authTeachers/Show.tsx

```diff
@@ -28,35 +53,95 @@ const AuthorizedTeacherShow = () => {
         } catch (e: any) {
             notify(e.message, { type: 'error' });
         }
-        setLoading(false);
+        setLoading(true);
         refresh();
     };
 
     return (
         <Show>
-            <SimpleShowLayout>
-                <TextField source="id" />
-                <EmailField source="email" />
-                <TextField source="userName" label="Name" />
-                <BooleanField source="created" looseValue />
-                <TextField source="branch" />
-                <WithRecord
-                    render={(record: AuthorizedTeacher) =>
-                        !record?.created ? (
-                            <LoadingButton
-                                variant="contained"
-                                color="primary"
-                                loading={loading}
-                                onClick={() => handleCreation(record)}
-                            >
-                                Create Account
-                            </LoadingButton>
-                        ) : (
-                            <></>
-                        )
-                    }
-                />
-            </SimpleShowLayout>
+            {loading ? (
```

Comment:
instead of ternary try to give something like

if(loading) return <>Loading..</>

return <>Other component</>